### PR TITLE
require a .gitignore file when we're in a git repo

### PR DESCRIPTION
**Pull request by JensTimmerman**

### CODE REVIEW

**Comment by stdweird** [comment]
version bump? and any minimal set of patterns?

File: lib/vsc/install/shared_setup.py [hunk 1]
```diff
@@ -296,19 +296,30 @@ def get_name_url(filename=None, version=None, license_name=None):
...
+    res = [os.path.relpath(p, base_dir) for p in paths]
 
     # primitive gitignore
-    gitignore = os.path.join(REPO_BASE_DIR, '.gitignore')
+    gitignore = os.path.join(base_dir, '.gitignore')
     if os.path.isfile(gitignore):
         patterns = [l.strip().replace('*','.*') for l in open(gitignore).readlines() if l.startswith('*')]
         reg = re.compile('^('+'|'.join(patterns)+')$')
+        # check if we at least filter out .pyc files, since we're in a python project
+        if not reg.search('bla.pyc'):
```
Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: list of patterns to test form a list, and also test `bla~` and `bla.pyo`

File: test/shared_setup.py [hunk 2]
```diff
@@ -65,3 +66,23 @@ def test_sanitize(self):
...
+
+
+    def test_rel_gitignore(self):
+        """
+        Test the rel_gitignore function
+        it should fail, since we don't specify a path for .pyc files in our gitignore
+        """
+        # when testing with a base_dir that has a .git folder, and a .gitignore file, we should get an error mentioning
+        # .pyc
+        base_dir = os.path.join(os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe()))), './testdata')
```
Comment (stdweird) [inline test/shared_setup.py]: ?

**Comment by JensTimmerman** [comment]
@stdweird remarks addressed

File: README.md [hunk 3]
```diff
@@ -68,7 +68,7 @@ Fix failing tests
 * Fix license headers as described in https://github.com/hpcugent/vsc-install/blob/master/lib/vsc/install/headers.py
   ```
```
Comment (stdweird) [inline README.md]: renders really ungly/unreadable
also, add something like 
```bash
find ./{lib,test} -type f -name '*.py' | xargs -I '{}' REPO_BASE_DIR=$PWD python -m vsc.install.headers '{}'
find ./bin -type f -name '*.py' | xargs -I '{}' REPO_BASE_DIR=$PWD python -m vsc.install.headers '{}' 1

```

File: lib/vsc/install/shared_setup.py [hunk 4]
```diff
@@ -296,19 +298,30 @@ def get_name_url(filename=None, version=None, license_name=None):
...
     if os.path.isfile(gitignore):
         patterns = [l.strip().replace('*','.*') for l in open(gitignore).readlines() if l.startswith('*')]
         reg = re.compile('^('+'|'.join(patterns)+')$')
+        # check if we at least filter out .pyc files, since we're in a python project
+        if not all([reg.search(text) for text in ['bla%s' % pattern for pattern in GITIGNORE_PATTERNS]]):
+            raise Exception("%s/.gitignore does not contain these patterns: %s " % (base_dir, GITIGNORE_PATTERNS))
+
         res = [f for f in res if not reg.search(f)]
+
+    elif os.path.isdir(os.path.join(base_dir, '.git')):
```
Comment (wpoely86) [inline lib/vsc/install/shared_setup.py]: change `elif` to `if`? You can have a `.gitignore` without being a git repo.

Comment (wpoely86) [inline lib/vsc/install/shared_setup.py]: `if not` I mean

Comment (JensTimmerman) [inline lib/vsc/install/shared_setup.py]: it's fine ot have a .gitignore and not be in a git repo, so we shouldn't raise exceptions then?

**Pull request merged**